### PR TITLE
Added distinction between transparent and opaque composition.

### PR DESCRIPTION
**Pull request by degregat**

### CODE REVIEW

File: src/architecture/basic-types/identity.md [hunk 1]
```diff
@@ -34,17 +34,35 @@ The canonical representation of an external identity is defined as `hash(verify,
...
 The canonical representation of composed external identities is defined the same as above, just with the new `verify` and `encrypt` functions.
 
-Compositions can be chained to create arbitrary combinations. For example, "threshold" identities, e.g. a 2-of-3 between `a`, `b`, and `c`, can be obtained with `(a && c) || (a && b) || (b && c)`. Composition of identities is used throughout the protocol and implementations may substitute more efficient operational representations where appropriate.
+Compositions can be chained to create arbitrary combinations. For example, "threshold" identities, e.g. a 2-of-3 between `a`, `b`, and `c`, can be obtained with `(a && c) || (a && b) || (b && c)`. Composition of identities is used throughout the protocol and implementations may substitute more efficient operational representations (e.g. different cryptographic schemes) where appropriate.
+
+In this transparent, non-interactive example (de-)composition information is available by construction.
+
+### Opaque Example
+
+A group of agents can also compose an opaque Identity interactively, s.t. composition information is not available to the outside.
```
Comment (cwgoes) [inline src/architecture/basic-types/identity.md:53]:
```suggestion
A group of agents can also compose an opaque identity interactively, s.t. composition information is not available to the outside.
```

File: src/architecture/basic-types/identity.md [hunk 2]
```diff
@@ -34,17 +34,35 @@ The canonical representation of an external identity is defined as `hash(verify,
...
+
+One example would be a using a Threshold cryptosystem and Distributed key generation [e.g. Threshold RSA](https://iacr.org/archive/eurocrypt2001/20450151.pdf).
```
Comment (cwgoes) [inline src/architecture/basic-types/identity.md:55]:
```suggestion
One example would be using distributed key generation and a threshold cryptosystem [e.g. Threshold RSA](https://iacr.org/archive/eurocrypt2001/20450151.pdf).
```

File: src/architecture/basic-types/identity.md [hunk 3]
```diff
@@ -34,17 +34,35 @@ The canonical representation of an external identity is defined as `hash(verify,
...
+
+Here the agents compute _one_ RSA keypair together, with only shares of the private key being generated by each agent. Decryption of messages encrypted to the single public key then requires cooperation of a subset of agents holding key shares, fulfilling the threshold requirements.
+
+To enable this setup, the `External Identity` needs to contain any address messages can be sent to, and the group needs to have a mechanism in place to do group internal routing, starting at that address.
```
Comment (cwgoes) [inline src/architecture/basic-types/identity.md:59]: Which address do you mean, exactly? Seems to me like `verify` and `encrypt` could both use the threshold public key. Do you mean the physical address? This is not (currently) part of the definition of an identity, I think we want to leave that as side information - of course, the group will need to _publish_ some physical address routes at which it (or a member) can be reached periodically.

File: src/architecture/basic-types/identity.md [hunk 4]
```diff
@@ -34,17 +34,35 @@ The canonical representation of an external identity is defined as `hash(verify,
...
+To observers outside of this group, will see only a single `External Identity` with an Address and a regular RSA public key.
+
+Optionally, the group can (selectively) supply decomposition information to make the composition transparent, e.g. routing information for constituent agents and information on the cryptographic scheme being used.
```
Comment (cwgoes) [inline src/architecture/basic-types/identity.md:62]: What sort of cryptographic information would be relevant here?

I think the provided routing information could (as long as revealing this info is acceptable) often be dynamic, e.g. "try physical address X first; if it doesn't work try Y", etc.

File: src/architecture/basic-types/identity.md [hunk 5]
```diff
@@ -34,17 +34,35 @@ The canonical representation of an external identity is defined as `hash(verify,
...
 
-> TODO: Do more efficient operational representations require different canonical serialisations? Can we make this nicely abstract still?
+> TODO: Do more efficient operational representations require different canonical serialisations? Can we make this nicely abstract still? 
+> Note: Ideally we would overload existing transparent operators whenever non-default cryptographic operations need to be used. This should be stored as side information with the respective Identity.
```
Comment (cwgoes) [inline src/architecture/basic-types/identity.md:65]: Not sure that I understand this exactly - `encrypt` and `verify` are just functions - what overloading are we talking about?

**Review by cwgoes — CHANGES_REQUESTED** [review]
Thanks - this seems like a very useful distinction.

File: src/architecture/basic-types/identity.md (hunk 3 above)
Comment (degregat) [reply src/architecture/basic-types/identity.md:59]: Yeah, I mean a physical address. I can open a section on side information to explain the connections.

File: src/architecture/basic-types/identity.md (hunk 5 above)
Comment (degregat) [reply src/architecture/basic-types/identity.md:65]: I meant overloading the `&&` and `||`, will clarify.

File: src/architecture/basic-types/identity.md (hunk 4 above)
Comment (cwgoes) [reply src/architecture/basic-types/identity.md:62]: A specific example of capabilities would be helpful for me, I guess - what couldn't just be encapsulated in the definitions of `encrypt` and `verify` ?

File: src/architecture/basic-types/identity.md (hunk 5 above)
Comment (cwgoes) [reply src/architecture/basic-types/identity.md:65]: Ah - but when would those operators be used, in the case of non-default composition? As an external party, I don't have the information required to compose in a non-default way anyways.

**Pull request merged**